src/libraries/System.Private.CoreLib/src/System/Globalization/CompareOptions.cs

```diff
@@ -3,17 +3,73 @@
 
 namespace System.Globalization
 {
+    /// <summary>
+    /// Defines the string comparison options to use with <see cref="CompareInfo"/>.
+    /// </summary>
     [Flags]
     public enum CompareOptions
     {
+        /// <summary>
+        /// Indicates the default option settings for string comparisons
+        /// </summary>
         None = 0x00000000,
+
+        /// <summary>
+        /// Indicates that the string comparison must ignore case.
+        /// </summary>
         IgnoreCase = 0x00000001,
+
+        /// <summary>
+        /// Indicates that the string comparison must ignore nonspacing combining characters, such as diacritics.
+        /// The <see href="https://go.microsoft.com/fwlink/?linkid=37123">Unicode Standard</see> defines combining characters as
+        /// characters that are combined with base characters to produce a new character. Nonspacing combining characters do not
+        /// occupy a spacing position by themselves when rendered.
+        /// </summary>
         IgnoreNonSpace = 0x00000002,
+
+        /// <summary>
+        /// Indicates that the string comparison must ignore symbols, such as white-space characters, punctuation, currency symbols,
+        /// the percent sign, mathematical symbols, the ampersand, and so on.
+        /// </summary>
         IgnoreSymbols = 0x00000004,
+
+        /// <summary>
+        /// Indicates that the string comparison must ignore the Kana type. Kana type refers to Japanese hiragana and katakana characters, which represent phonetic sounds in the Japanese language.
+        /// Hiragana is used for native Japanese expressions and words, while katakana is used for words borrowed from other languages, such as "computer" or "Internet".
+        /// A phonetic sound can be expressed in both hiragana and katakana. If this value is selected, the hiragana character for one sound is considered equal to the katakana character for the same sound.
+        /// </summary>
         IgnoreKanaType = 0x00000008,
+
+        /// <summary>
+        /// Indicates that the string comparison must ignore the character width. For example, Japanese katakana characters can be written as full-width or half-width.
+        /// If this value is selected, the katakana characters written as full-width are considered equal to the same characters written as half-width.
+        /// </summary>
         IgnoreWidth = 0x00000010,
+
+        /// <summary>
+        /// Indicates that the string comparison must sort sequences of digits (Unicode general category "Nd") based on their numeric value.
+        /// For example, "2" comes before "10". Non-digit characters such as decimal points, minus or plus signs, etc.
+        /// are not considered as part of the sequence and will terminate it.
+        /// </summary>
```

Comment:
>  /// </summary> [](http://example.com/codeflow?start=6&length=16)

will be good to add a remarks to this one giving more information like this option can be used in comparisons but not for search (IndexOf/StartsWith/EndsWith). Will be good to hint the behavior difference too when ICU is used against NLS. And last tell this option cannot be combined with the ordinal operations,

Reply:
The comment was getting long so I just included just the part about indexing. Mentioning search might be confusing because people might consider IndexOf (which isn't supported) as search and GetHashCode (which is supported) as not. I think it's easier to just say that NumericOrdering works in all cases except for indexing.

I prefer to keep the combination behavior with Ordinal and OrdinalIgnoreCase on those members since this is really a property of theirs instead of numeric ordering. This is also consistent with the other options as well which don't mention Ordinal even though they can't combine with them either.

I think the ICU and NLS differences should probably go in docs rather than in doc comments since NLS usage is not going to be high. There are already docs about NLS/ICU differences that we can append to (https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#behavioral-differences).

Reply:
Well, if you prefer that, it will be better to edit the docs of the indexing APIs and add the remark there. I am trying to make it easy for the API users to understand when this new enum value is not allowed. I guess users can be puzzled if they get exceptions and do not understand what is wrong. 

You don't have to block the PR on that, but it will be good to open a doc issue/PR to add the info as needed.

Reply:
I like that idea. I just checked and we already list the valid options so I will update those docs once this goes in:
[Compare](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.compareinfo.compare?view=net-8.0#system-globalization-compareinfo-compare(system-string-system-string-system-globalization-compareoptions))
[IndexOf](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.compareinfo.indexof?view=net-8.0#system-globalization-compareinfo-indexof(system-string-system-char-system-int32-system-globalization-compareoptions))